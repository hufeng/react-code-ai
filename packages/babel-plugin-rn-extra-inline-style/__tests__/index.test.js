const babel = require('babel-core');
const extraInlineStyle = require('../index');

it('extra inline stle without import StyleSheet', () => {
  const tpl = `
    const Hello = () => (
      <View style={{flex: 1, backgroundColor: 'red'}}>
        <Text style={styles.text}>hello world</Text>      
      </View>
    );
  `;

  const { code } = babel.transform(tpl, {
    plugins: [extraInlineStyle, 'syntax-jsx']
  });

  expect(code).toMatchSnapshot();
});

it('extra inline style with import StyleSheet', () => {
  const tpl = `
  import {StyleSheet} from 'react-native';

  const Hello = () => (
    <View style={{flex: 1, backgroundColor: 'red'}}>
      <Text style={styles.text}>hello world</Text>      
    </View>
  );
`;

  const { code } = babel.transform(tpl, {
    plugins: [extraInlineStyle, 'syntax-jsx']
  });

  expect(code).toMatchSnapshot();
});
