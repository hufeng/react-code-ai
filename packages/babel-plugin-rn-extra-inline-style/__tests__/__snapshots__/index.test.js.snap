// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extra inline stle without import StyleSheet 1`] = `
"import { StyleSheet as _StyleSheet } from 'react-native';

const Hello = () => <View style={ai.style1}>
        <Text style={styles.text}>hello world</Text>      
      </View>;

const ai = _StyleSheet.create({
  style1: { flex: 1, backgroundColor: 'red' }
});"
`;

exports[`extra inline style with import StyleSheet 1`] = `
"
import { StyleSheet } from 'react-native';

const Hello = () => <View style={ai.style1}>
      <Text style={styles.text}>hello world</Text>      
    </View>;
const ai = StyleSheet.create({
  style1: { flex: 1, backgroundColor: 'red' }
});"
`;

exports[`extra inline style with import StyleSheet and multiple style 1`] = `
"
import { StyleSheet } from 'react-native';

const Hello = () => <View style={ai.style1}>
      <Text style={styles.text}>hello world</Text>      
      <Text style={[styles.text, ai.style2]}>hello world</Text>      
    </View>;
const ai = StyleSheet.create({
  style1: { flex: 1, backgroundColor: 'red' },
  style2: { color: 'yellow', fontSize: 12 }
});"
`;
